src/caf/toolkit/translation.py

```diff
@@ -625,22 +626,15 @@ def pandas_matrix_zone_translation(
         dataframe_name="col_translation",
     )
 
-    # Check the matrix and translation dtypes match
-    if matrix.index.dtype != row_translation[translation_from_col].dtype:
-        raise ValueError(
-            "dtypes of `matrix.index` and `translation` in `from_zone_col` "
-            "must match.\n"
-            f"Matrix index data type: {matrix.index.dtype}\n"
-            f"Row Translation data type: {row_translation[translation_from_col].dtype}"
-        )
-
-    if matrix.columns.dtype != col_translation[translation_from_col].dtype:
-        raise ValueError(
-            "dtypes of `matrix.columns` and `col_translation` in `from_zone_col` "
-            "must match.\n"
-            f"Matrix column Dtype: {matrix.columns.dtype}\n"
-            f"Col Translation data type: {col_translation[translation_from_col].dtype}"
-        )
+    # Set the dtypes to match
+    matrix.index, row_translation[translation_from_col] = pd_utils.cast_to_common_type(
+        matrix.index,
+        row_translation[translation_from_col],
+    )
+    matrix.index, col_translation[translation_from_col] = pd_utils.cast_to_common_type(
+        matrix.index,
+        col_translation[translation_from_col],
+    )
```

Comment:
This should be comparing the matrix columns not index.
```suggestion
    matrix.columns, col_translation[translation_from_col] = pd_utils.cast_to_common_type(
        matrix.columns,
        col_translation[translation_from_col],
    )
```